src/client/Microsoft.Identity.Client/Instance/Authority.cs

```diff
@@ -208,6 +218,22 @@ private static void ValidateSameHost(AuthorityInfo requestAuthorityInfo, Authori
             }
         }
 
+        private static bool IsAuthorityAliased(RequestContext requestContext, AuthorityInfo requestAuthorityInfo)
```

Comment:
This verification is expensive, as it involves making a network call. We should only do it if the Config authority and the Request authority are different.